invenio_app_ils/permissions.py

```diff
@@ -79,6 +97,53 @@ def files_permission(obj, action=None):
     return backoffice_permission()
 
 
+def is_loan_owner_permission(patron_pid):
+    """Return permission to allow loan owner, librarians and admins."""
+    return Permission(UserNeed(int(patron_pid)), backoffice_access_action)
+
+
+def loan_extend_circulation_permission(loan):
+    """Return permission to allow only owner and librarians to extend loan."""
+    if not current_user or not current_user.id:
+        abort(401)
+
+    if current_user.id == int(loan["patron_pid"]):
+        validate_automatic_loan_extension(loan)
+
+    return is_loan_owner_permission(loan["patron_pid"])
+
+
+def validate_automatic_loan_extension(loan):
+    """Validate if the loan can be extended automatically."""
+    # NOTE: Validate max extensions
+    if not (
+        loan["extension_count"] <=
```

Comment:
This is already checked by the transition itself. Do we keep it to be checked again?
If we keep it: `loan["extension_count"]` -> `loan.get("extension_count", 0)`
WDYT?